### ISSUE TITLE
Cannot install on windows machine

### ISSUE THREAD

**Issue by ydsood** [ritz078/rollup-plugin-filesize#33]
**Summary**
Trying to install the package for a rollup build on a machine with only node and npm installed. I get an error that prevents installation.

Here is the detailed error

```shell
C:\Users\ysood\Source\Sandbox\redux-form-datagrid>npm install --save rollup-plugin-filesize

> iltorb@2.4.1 install C:\Users\ysood\Source\Sandbox\redux-form-datagrid\node_modules\iltorb
> node ./scripts/install.js || node-gyp rebuild

info looking for cached prebuild @ C:\Users\ysood\AppData\Roaming\npm-cache\_prebuilds\45bfec-iltorb-v2.4.1-node-v64-win32-x64.tar.gz
http request GET https://github.com/MayhemYDG/iltorb/releases/download/v2.4.1/iltorb-v2.4.1-node-v64-win32-x64.tar.gz
WARN install unable to get local issuer certificate

C:\Users\ysood\Source\Sandbox\redux-form-datagrid\node_modules\iltorb>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:154:21)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\ysood\Source\Sandbox\redux-form-datagrid\node_modules\iltorb
gyp ERR! node -v v10.13.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN redux-form-datagrid@0.1.3 requires a peer of react-redux@5.x but none is installed. You must install peer dependencies yourself.
npm WARN redux-form-datagrid@0.1.3 requires a peer of redux@4.x but none is installed. You must install peer dependencies yourself.
npm WARN redux-form-datagrid@0.1.3 requires a peer of redux-form@7.x but none is installed. You must install peer dependencies yourself.
npm WARN redux-form-datagrid@0.1.3 requires a peer of semantic-ui-css@2.x but none is installed. You must install peer dependencies yourself.
npm WARN @babel/plugin-syntax-object-rest-spread@7.0.0 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! iltorb@2.4.1 install: `node ./scripts/install.js || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the iltorb@2.4.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ysood\AppData\Roaming\npm-cache\_logs\2018-11-15T15_53_54_364Z-debug.log
```
Appears that one of the dependencies needs python and is somehow also failing to find some _local issuer certificate_ . Not sure if it's a good idea for a node package to depend on a completely different technology.

Sadly having to remove this for now from my package as it is preventing contributors from working on it. Will add once fixed.

**Comment by dosentmatter** [comment]
It's because of PR #22 that added `brotli-size`. `brotli-size` depends on `iltorb` which uses `node-gyp` for native bindings. `node-gyp` requires `python`, `make`, and C/C++ compiler. `iltorb` runs `node-gyp` and install script after `npm install`. There are a couple of options and they can be mix and matched:

1. Move `brotli-size` to `optionalDependencies` in package.json. `showBrotliSize` is `false` by default so it shouldn't be a requirement to install this package.
2. Use [`brotli-fsize`](https://www.npmjs.com/package/brotli-fsize). This package uses node@11.7.0 native brotli support. So it has to have a backup for users on older node versions.
3. Use [`brotli`](https://www.npmjs.com/package/brotli), a JS port of brotli. You would have to use the brotli API directly and write custom code to get the file size. Should be easy. The code for [`brotli-size`](https://github.com/erwinmombay/brotli-size/blob/master/index.js) and [`brotli-fsize`](https://github.com/Developmint/brotli-fsize/blob/master/index.js) are really short. They can be ported to use brotli.js.

You can also use all of them by having `brotli-fsize` and `brotli-size` in `optionalDependencies` and `brotli` in `dependencies`. Then in code, check for `brotli-fsize` (node@11.7), then `brotli-size` (iltorb), and finally fall back to `brotli` (JS).

If you want to optionally use `iltorb`, you can add `maybe-iltorb` to `optionalDependencies`. It skips installation if you are on node@11.7+ (by checking that zlib has brotli supprt).

Anyway I think this should be fixed because, I can't really use this plugin in my build pipeline that builds with a node docker container. I would have to install the node-gyp dependencies.

**Comment by dosentmatter** [comment]
@ydsood, to make it work, you have two options:

1. Downgrade to rollup-plugin-filesize@2.0.0, before `brotli-size` was added.
2. Install `python`, `make`, and `gcc` on your computer for `node-gyp` to work.

**Comment by dosentmatter** [comment]
I think this can be fixed by updating to a later version of `brotli-size`. `rollup-plugin-filesize` is currently fixed at [`brotli-size@0.1.0`](https://github.com/ritz078/rollup-plugin-filesize/blob/c5899a1784f9efbf219099b766bc6bf2a72e04b8/package.json#L33).

`brotli-size` removed the `iltorb` dependency. It now uses native `brotli` of [`node >=10.16`](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2019-05-28-version-10160-dubnium-lts-bethgriggs).

if you want it to work on `node < 10.16`, `brotli-size` can still be an optionalDependency.
